src/common/libflux/plugin.c

```diff
@@ -226,6 +226,18 @@ int flux_plugin_set_conf (flux_plugin_t *p, const char *json_str)
     return 0;
 }
 
+int flux_plugin_get_conf (flux_plugin_t *p, char **rp)
```

Comment:
Would it make sense to cache the string-encoded `p->conf` in the `flux_plugin_t` so that it doesn't have to be dumped on each call (and the user doesn't have to free the resuilt)?

Reply:
I thought about that, but considered it a premature optimization at the time (IIRC). Now that you mention it again though, it does seem like a good usability improvement.

Reply:
Ah, I remember why I didn't do this at first. The cached value will have to be invalidated for every call to `flux_plugin_set_conf(3)`. This was abject laziness on my part. Forgiveness please.